src/OrchardCore/OrchardCore.Setup.Abstractions/ISetupUserIdGenerator.cs

```diff
@@ -0,0 +1,7 @@
+namespace OrchardCore.Setup.Services
+{
+    public interface ISetupUserIdGenerator
```

Comment:
To avoid a reference between `Setup` and `Users` we have two id generators, one for during setup (because the user is not created until after the recipe is run - i.e. the user creation is performed through setup event handlers) and one for user creation, after setup